### ISSUE TITLE
Domain integration includes cells in overlap

### ISSUE THREAD

**Issue by baagaard-usgs** [geodynamics/pylith#528]
## Describe the bug

The domain integration includes cells in the overlap. As a result, it looks like we get contributions from both processes.

## Expected behavior

Cells in the overlap are not supposed to be in the material-id labels. If the cells are not in the material-id labels, then they won't be included in the integration.

## How to reproduce

Run any problem with a fault in parallel.

**PETSc branch**: `knepley/pylith`
**PyLith branch**:  `baagaard/feature-version-3.0.3`

```
cd examples/strikeslip-2d
pylith step01_slip.cfg --nodes=2
```

**Comment by knepley** [comment]
@baagaard-usgs I have remove the cells and the problem appears to run for me. I pushed to your branch

**Comment by baagaard-usgs** [comment]
@knepley I don't see any change in the behavior. I still end up with duplicate integration. Running on 2 processes in the debugger, the code in `pylith::topology::_Distributor::distributeOverlap()` never encounters the condition loc >= 0 and executes `DMLabelClearValue()`.

```
cd examples/strikeslip-2d
pylith step01_slip.cfg --nodes=2 --petsc.start_in_debugger
```

There are 715 cells for material-id=1 and 715 cells for material-id=2. For material-id=1, process 0 integrates over 389 cells and process 1 integrates over 329 cells (total = 718). Break at line 416 in IntegratorDomain.cc and print dsLabel.

**Comment by knepley** [comment]
@baagaard-usgs Did you run after I fixed the mesh I was pointing to? Second push

**Comment by knepley** [comment]
I ran np=1 to 8 and got no faults.

**Comment by knepley** [comment]
It looks like one of those arrays is NULL (since you get 0x1), probably u or coefficients. It should be straightforward to track back to the original NULL.

**Comment by baagaard-usgs** [comment]
Are you using PETSc `knepley/pylith` or `main`?

**Comment by baagaard-usgs** [comment]
coefficients is `0x0000000000000001`. Going up the stack, I see a[3] is

```
(PetscScalar *[3]) $9 = {
  [0] = 0x0000000000000001
  [1] = 0x0000000500000000
  [2] = 0x0000000166d49850
}
```

I think removing the overlap from the material-id may have messed up the hybrid integration where we are trying to get the auxiliary field for the adjacent cell (in the overlap).

**Comment by knepley** [comment]
> Are you using PETSc `knepley/pylith` or `main`?

I am on knepley/pylith

**Comment by baagaard-usgs** [comment]
I initialized a[s] to NULL immediately before getting the work array. I still get the same segmentation fault. I think it is trying to use a[0] but it doesn't have it.

**Comment by baagaard-usgs** [comment]
The correct fix is to drop cells in the overlap from the integration over domain cells. We need the overlap in the label so that we have auxiliary data for the integration over the interface (fault) cells.

Fixed in `baagaard/feature-version-3.0.3`.
